gammapy/background/models.py

```diff
@@ -71,3 +83,699 @@ def evaluate(self, x, y):
+        imhdu.header['CTYPE3'] = 'ENERGY'
+        imhdu.header['CUNIT3'] = '{0.unit:FITS}'.format(self.energy_bins)
+        imhdu.header['E_THRES'] = self.energy_bins[0].value
+
+        # get WCS object and write it out as a FITS header
+        wcs_header = self.det_wcs.to_header()
+
+        # get energy values as a table HDU, via an astropy table
+        energy_table = Table()
+        energy_table['ENERGY_BIN_EDGES'] = self.energy_bins
+        energy_table.meta['name'] = 'ENERGY_BINS'
+
+        enhdu = table_to_fits_bin_table(energy_table)
+
+        hdu_list = fits.HDUList([imhdu, enhdu])
+
+        return hdu_list
+
+    def write(self, outfile, format='table', **kwargs):
+        """Write cube background model to fits file.
+
+        Several output formats are accepted, depending on the value
+        of the `format` parameter:
+
+        * bin_table (default and preferred format): `~astropy.io.fits.BinTableHDU`
+        * image (alternative format): `~astropy.io.fits.PrimaryHDU`,
+          with the energy binning stored as `~astropy.io.fits.BinTableHDU`
+
+        Depending on the value of the `format` parameter, this
+        method calls either `~astropy.io.fits.BinTableHDU.writeto` or
+        `~astropy.io.fits.HDUList.writeto`, forwarding the
+        `kwargs` arguments.
+
+        Parameters
+        ----------
+        outfile : str
+            name of file to write
+        format : str, optional
+            format of the bg cube to write
+        kwargs
+            extra arguments for the corresponding `io.fits` `writeto` method
+        """
+        if format == 'table':
+            self.to_fits_bin_table().writeto(outfile, **kwargs)
+        elif format == 'image':
+            self.to_fits_image().writeto(outfile, **kwargs)
+        else:
+            raise ValueError("Invalid format {}.".format(format))
+
+    @property
+    def image_extent(self):
+        """Image extent `(x_lo, x_hi, y_lo, y_hi)`.
+
+        Returns
+        -------
+        im_extent : `~astropy.coordinates.Angle`
+            array of bins with the image extent
+        """
+        bx = self.detx_bins
+        by = self.dety_bins
+        return Angle([bx[0], bx[-1], by[0], by[-1]])
+
+    @property
+    def spectrum_extent(self):
+        """Spectrum extent `(e_lo, e_hi)`.
+
+        Returns
+        -------
+        spec_extent : `~astropy.units.Quantity`
+            array of bins with the spectrum extent
+        """
+        b = self.energy_bins
+        return Quantity([b[0], b[-1]])
+
+    @property
+    def image_bin_centers(self):
+        """Image bin centers `(x, y)`.
+
+        Returns
+        -------
+        detx_edges_centers : `~astropy.coordinates.Angle`
+            array of image bin centers (X coord)
+        dety_edges_centers : `~astropy.coordinates.Angle`
+            array of image bin centers (Y coord)
+        """
+        detx_edges_centers = self.detx_bins[:-1] + np.diff(self.detx_bins)/2.
```

Comment:
Thinking about this some more, I think this is a tick more readable:
```python
bin_centers = 0.5 * (bin_edges[:-1] + bin_edges[-1:])
```
and is more similar to the expression for log bin centers (geometric mean):
```
bin_centers = np.sqrt(bin_edges[:-1] * bin_edges[1:])
```
So please use that.

It's also faster! :speedboat: :racehorse: 
:-)